third-party/gasnet/Makefile

```diff
@@ -116,6 +116,9 @@ GASNET_CFLAGS += -fPIE
 endif
 endif
 
+# workaround for https://gasnet-bugs.lbl.gov/bugzilla/show_bug.cgi?id=4787
+GASNET_CFLAGS += -Wno-incompatible-pointer-types
```

Comment:
It seems odd to hard-code use of a Gnu option here, without verifying the compiler actually IS Gnu. IOW I think this only works for your supported target compilers because LLVM/Apple clang both happen to also accept an option of the same name.

Perhaps this should be conditional on `CHPL_MAKE_TARGET_COMPILER` similar to line 112 above?

Reply:
I agree with @bonachea that this should not be assumed valid for all compilers.
At _least_ the Intel "Classic" (non-llvm) compilers are not happy with it:
```
{phargrov@pcp-d-5 ~}$ icc -Wno-incompatible-pointer-types Hello/hello.c
icc: command line warning #10148: option '-Wno-incompatible-pointer-types' not supported
```